volatility3/plugins/windows/notepad.py

```diff
@@ -0,0 +1,707 @@
+                     0x7f75, 0x82ad, 0x831c, 0x83c5, 0x851a, 0x8599, 0x8654, 0x8717, 0x8852, 0x8a6b, 0x8c2d, 0x8d0a,
+                     0x8d2c, 0x8d81, 0x8e99, 0x90b8, 0x9149, 0x9163, 0x9295, 0x94f0, 0x95ef, 0x9611, 0x9688, 0x9798,
+                     0x97a0, 0x9897, 0x98f4, 0x9952, 0x9985, 0x9b51, 0x9e70, 0x9e92, 0xac94, 0xb51c, 0xb6ab, 0xb810,
+                     0xc385, 0xc583, 0xc660, 0xc6f0, 0xd5d0, 0xd6d7, 0xe035, 0xfbde, 0xff96, 0x6, 0x325, 0x32e, 0x33f,
+                     0x358, 0x38c, 0x5bc, 0x69a, 0x718, 0x721, 0x729, 0x72a, 0x787, 0x905, 0x916, 0x985, 0x9c0, 0xaaf,
+                     0xac0, 0xb15, 0xd2d, 0xf55, 0xf66, 0x120e, 0x12eb, 0x179a, 0x1ebd, 0x2000, 0x2008, 0x2198, 0x2228,
+                     0x263e, 0x4e1f, 0x4e52, 0x4fae, 0x4fb6, 0x50b5, 0x51bd, 0x51d6, 0x51e7, 0x524e, 0x5289, 0x5319,
+                     0x53ee, 0x5443, 0x57ae, 0x592d, 0x5993, 0x5b5f, 0x5d4c, 0x5dba, 0x5f27, 0x5fd1, 0x606b, 0x61fa,
+                     0x62df, 0x63c4, 0x652a, 0x6727, 0x67d1, 0x67e2, 0x693f, 0x6b93, 0x6bdf, 0x6c9b, 0x6cde, 0x70af,
+                     0x70d9, 0x72e0, 0x72f9, 0x7360, 0x7368, 0x74f7, 0x766e, 0x7690, 0x76b1, 0x76ba, 0x7852, 0x7940,
+                     0x7d0a, 0x7f87, 0x7fa8, 0x814a, 0x852c, 0x863c, 0x87ba, 0x88f4, 0x8ba5, 0x8bb6, 0x8e81, 0x94b5,
+                     0x9890, 0x9a6d, 0x9a7e, 0x9a8f, 0x9aa7, 0x9ab8, 0x9ad2, 0x9ae3, 0x9af4, 0x9c48, 0x9ed6, 0x9f2c,
+                     0xaf2c, 0xaf34, 0xb080, 0xb220, 0xb550, 0xb69c, 0xbc1f, 0xd329}
+
+
+class Notepad(interfaces.plugins.PluginInterface):
+    """Dumps displayed text from open Notepad processes"""
+
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.VersionRequirement(name="pslist", component=pslist.PsList, version=(2, 0, 0)),
+            requirements.VersionRequirement(name="vadinfo", component=vadinfo.VadInfo, version=(2, 0, 0)),
+            requirements.ListRequirement(
+                name="pid",
+                element_type=int,
+                description="Manually specifies notepad process IDs",
+                optional=True,
+            ),
+        ]
+
+    def _generator(self, procs):
+        kernel = self.context.modules[self.config["kernel"]]
+
+        protect_values = vadinfo.VadInfo.protect_values(
+            self.context,
+            kernel.layer_name,
+            kernel.symbol_table_name
+        )
+
+        for proc in procs:
+            proc_id = "Unknown"
+            try:
+                proc_id = proc.UniqueProcessId
+                proc_layer_name = proc.add_process_layer()
```

Comment:
I'd prefer proc_layer_name being given a bogus default initalization, much like `proc_id` above, although `proc_id` above should initialize to a number or `None`.  To support the `unknown` name, change the debug message to be `proc_id or 'Unknown'` although note that if `proc_id` is then 0 this will result in the layer being referred to as Unknown.